test/integration/multi-arch/compute.cpp

```diff
@@ -0,0 +1,19 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#include <eve/wide.hpp>
+#include <eve/function/sqrt.hpp>
+#include <eve/function/store.hpp>
+#include "eve/arch/spec.hpp"
+
+void compute(float* data, eve::current_api_type)
```

Comment:
I don't think this is how you do it because what you want is a staticly initialized variable.
Which means - all of these should have the same type.

Reply:
this exact .cpp is compiled N times with different compiler options, thus for each one, this type changes.
That's the job of the eve_variants cmake function.

Reply:
So each .a has a different prototype, those listed in the main.cpp
When the main call it, it uses the runtime detector to find the best match and call the appropriate symbol.